Capture output goes to terminal by default

tinvaan opened the pull request:
Capture and console.capture() accepts a kwargs, where you may specify echo=False to silence output being sent to terminal.

## Type of changes

- [x] Bug fix
- [x] New feature
- [ ] Documentation / docstrings
- [x] Tests
- [ ] Other

## Checklist

- [x] I've run the latest [black](https://github.com/psf/black) with default args on new code.
- [ ] I've updated CHANGELOG.md and CONTRIBUTORS.md where appropriate.
- [x] I've added tests for new code.
- [x] I accept that @willmcgugan may be pedantic in the code review.

## Description

Attempts a fix for https://github.com/Textualize/rich/issues/2172 by implementing @willmcgugan's [suggestion](https://github.com/Textualize/rich/issues/2172#issuecomment-1154942255)

willmcgugan commented on rich/console.py:
```diff
@@ -327,9 +327,10 @@ class Capture:
     """
 
-    def __init__(self, console: "Console") -> None:
+    def __init__(self, console: "Console", **kwargs) -> None:
```
Don't see any reason to use kwargs here. It breaks typing.

tinvaan replied:
hmm guessing you're suggesting a default arg instead - `def __init__(..., echo: bool=True)`?

willmcgugan commented on rich/console.py:
```diff
@@ -1080,7 +1083,7 @@ def bell(self) -> None:
         self.control(Control.bell())
 
-    def capture(self) -> Capture:
+    def capture(self, **kwargs) -> Capture:
```
No need for `kwargs`.

willmcgugan commented on rich/console.py:
```diff
@@ -342,6 +343,8 @@ def __exit__(
     ) -> None:
         self._result = self._console.end_capture()
+        if self._echo:
+            self._console.out(self._result)
```
This is going to run captured input through `print` again, which may modify the result.

tinvaan replied:
right, guess I'll invoke `print` directly then.

willmcgugan commented on tests/test_console.py:
```diff
@@ -379,22 +379,33 @@ def test_capture():
         console.print("Hello")
     assert capture.get() == "Hello\n"
 
+    with console.capture(echo=False) as capture:
```
This doesn't test that `echo` is working.

tinvaan commented:
@willmcgugan I've addressed the review comments, but I see locally that my change breaks the `pager` tests(`tests/test_console.py::test_pager`). 

However, printing the test variable to console shows the text styled as expected(**Hello World**) but we're asserting for a plain string in the tests. Shouldn't it be the other way round? What am I missing?

<img width="962" alt="Screenshot 2023-01-13 at 8 51 28 AM" src="https://user-images.githubusercontent.com/5815030/212229716-e3180715-6701-448c-bec9-69b8f4a2459c.png">

tinvaan commented:
@willmcgugan *poke*?

willmcgugan commented:
There are other considerations you may not be aware of. You might want to leave this one to us.

tinvaan commented:
> There are other considerations you may not be aware of. You might want to leave this one to us.

Ok, I'll keep the PR open for anyone who might want to attempt a fix. Thanks!
